File: backend/src/api/slug.ts
import { randomBytes } from "node:crypto";

/** Builds a URL-safe, unique-ish slug from a job title. */
export function slugify(title: string): string {
  const base =
    title
      .toLowerCase()
      .replace(/[^a-z0-9]+/g, "-")
      .replace(/(^-|-$)/g, "")
      .slice(0, 40) || "job";
  return `${base}-${randomBytes(3).toString("hex")}`;
}

File: backend-py/app/api/slug.py
"""URL-safe, unique-ish slug from a job title (port of slug.ts)."""

from __future__ import annotations

import re
import secrets


def slugify(title: str) -> str:
    base = title.lower()
    base = re.sub(r"[^a-z0-9]+", "-", base)
    base = re.sub(r"(^-|-$)", "", base)
    base = base[:40] or "job"
    return f"{base}-{secrets.token_hex(3)}"
